### PR TITLE
[Cache bolt] Decouple Transport & Domain layer, add unit testing.

### LINKED ISSUE #1102: Add unit tests for Cache topology

### PR DESCRIPTION
**Pull request by Settis**
Closes #1102 issue.

### CODE REVIEW

File: services/wfm/src/main/java/org/openkilda/wfm/topology/cache/service/CacheService.java [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+import org.openkilda.pce.cache.Cache;
+import org.openkilda.pce.cache.FlowCache;
+import org.openkilda.pce.cache.NetworkCache;
+import org.openkilda.pce.provider.PathComputer;
+import org.openkilda.wfm.Sender;
+import org.openkilda.wfm.share.utils.PathComputerFlowFetcher;
+import org.openkilda.wfm.topology.cache.StreamType;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.storm.state.InMemoryKeyValueState;
```
Comment (carmine) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/cache/service/CacheService.java]: Is this necessary? It creates a coupling with storm.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/cache/service/CacheService.java [hunk 2]
```diff
@@ -101,133 +76,76 @@
...
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void initState(InMemoryKeyValueState<String, Cache> state) {
-        this.state = state;
-
+    public CacheService(InMemoryKeyValueState<String, Cache> state, PathComputer pathComputer) {
```
Comment (surabujin) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/cache/service/CacheService.java]: Don't pass "state" here... if you are going to split "transport" and service layers.

PS This state is not needed even into storm's bolt... but its elimination is not part of this issue.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/cache/transport/CacheTopology.java [hunk 3]
```diff
@@ -87,8 +88,8 @@ public StormTopology createTopology() throws NameCollisionException {
         ComponentObject.serialized_java(org.apache.storm.utils.Utils.javaSerialize(pathComputerAuth));
         BoltDeclarer boltSetup = builder.setBolt(BOLT_ID_CACHE, cacheBolt, parallelism)
                 .shuffleGrouping(SPOUT_ID_COMMON)
-// (carmine) as per above comment, only a single input streamt
-//                .shuffleGrouping(SPOUT_ID_TOPOLOGY)
+        // (carmine) as per above comment, only a single input streamt
+        //                .shuffleGrouping(SPOUT_ID_TOPOLOGY)
         ;
```
Comment (abochkarev) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/cache/transport/CacheTopology.java]: It look like ';' is not on the right line.

File: services/wfm/src/test/java/org/openkilda/wfm/topology/cache/service/CacheServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+public class CacheServiceTest {
+
+    private static final String thirdFlowId = "third-flow";
```
Comment (abochkarev) [inline services/wfm/src/test/java/org/openkilda/wfm/topology/cache/service/CacheServiceTest.java]: Will it be possible to rename thirdFlowId to THIRD_FLOW_ID as it const?

File: services/wfm/src/test/java/org/openkilda/wfm/topology/cache/service/CacheServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    private static final ImmutablePair<Flow, Flow> thirdFlow = new ImmutablePair<>(
```
Comment (abochkarev) [inline services/wfm/src/test/java/org/openkilda/wfm/topology/cache/service/CacheServiceTest.java]: ditto

**Comment by Settis** [comment]
Sonar can check only non forked repositories. I've recreated this pull request with local branch #1176. Let's proceed with this new pull request.

**Pull request closed without merging**